core/peerset/src/lib.rs

```diff
@@ -434,16 +442,20 @@ impl Peerset {
 		self.alloc_slots();
 	}
 
-	/// Adds a discovered peer id to the PSM.
+	/// Adds discovered peer ids to the PSM.
 	///
 	/// > **Note**: There is no equivalent "expired" message, meaning that it is the responsibility
 	/// >			of the PSM to remove `PeerId`s that fail to dial too often.
-	pub fn discovered(&mut self, peer_id: PeerId) {
-		if self.data.in_slots.contains(&peer_id) || self.data.out_slots.contains(&peer_id) {
-			return;
+	pub fn discovered<I: IntoIterator<Item = PeerId>>(&mut self, peer_ids: I) {
+		for peer_id in peer_ids {
+			if !self.data.in_slots.contains(&peer_id) && !self.data.out_slots.contains(&peer_id) && !self.data.discovered.contains(&peer_id) {
```

Comment:
I added this additional check in `data.discovered`, so we can print `peer_id` without cloning it